Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -190,34 +206,52 @@ updateDriverWeeklyZscore ride rideDate weekStartDate weekEndDate driverZscore ri
             case driverZscore of
               Nothing -> weeklyLeaderBoardConfig.zScoreBase + getMeters (fromMaybe 0 rideChargeableDistance)
               Just zscore -> do
-                let (prevTotalRides, prevTotalDistance) = getRidesAndDistancefromZscore zscore weeklyLeaderBoardConfig.zScoreBase
-                let currTotalRides = prevTotalRides + 1
-                let currTotalDist = prevTotalDistance + fromMaybe 0 rideChargeableDistance
-                currTotalRides * weeklyLeaderBoardConfig.zScoreBase + getMeters currTotalDist
-      Hedis.zAddExp (makeWeeklyDriverLeaderBoardKey merchantId weekStartDate weekEndDate) ride.driverId.getId (fromIntegral currZscore) (lbExpiry.getSeconds)
+                let (prevTotalRides, prevTotalDistance, prevTotalValidCustomers, prevTotalReferrals) = getParamsfromZscore zscore weeklyLeaderBoardConfig.zScoreBase driverLeaderBoardType
+                case driverLeaderBoardType of
+                  LConfig.RIDE -> do
+                    let currTotalRides = prevTotalRides + 1
+                    let currTotalDist = prevTotalDistance + fromMaybe 0 rideChargeableDistance
+                    currTotalRides * weeklyLeaderBoardConfig.zScoreBase + getMeters currTotalDist
+                  LConfig.REFERRAL -> do
+                    let currTotalValidCustomers = prevTotalValidCustomers + (if validCustomer then 1 else 0)
+                    let currTotalReferrals = prevTotalReferrals + 1
```

Comment:
same

Reply:
we require to do the referral ranking according to the number of customers referred who has taken valid ride. That's why added valid customer check
first level sorting - total activated customers
second level sorting - total referred customers